### DISCUSSION TITLE
Using Isar with Freezed seems not compatible

### DISCUSSION THREAD

**Discussion by stevehayles** [isar/isar#518] [Help]
I'm trying to integrate an existing Freezed model and persist it using Isar but despite seeing some examples that suggest it works OK I am running into difficulties.  Using a fairly simple model with an added Id property as per the code below

`part 'location_model.g.dart';
part 'location_model.freezed.dart';

@freezed
@Collection()
class LocationModel with _$LocationModel {
  factory LocationModel({
    @Id() int? localId,
    required double latitude,
    required double longitude,
    required double accuracy,
    required double altitude,
    required double speed,
    required double speedAccuracy,
    required double heading,
    @Index(unique: true) required double time,
    required bool isMocked,
    required String provider,
  }) = _LocationModel;

Running the generator creates `the freezed classes but the Isar generator complains that there is "No int property named "id" or annotated with @Id()."

I'm using Isar 2.5.0 to avoid another similar problem that appears in more recent versions of Isar which use a typedef alias and the fact that freezed expands the typedef to the original type (that's not my question here though).  Has anyone got a working code example or any other advice on getting this working ?

**Comment by masreplay** [comment]
Hello, @stevehayles 🖐

here is a full article https://medium.com/@masreplay/integrating-freezed-with-isar-for-effortless-data-management-in-flutter-df03c3e37e59?source=user_profile---------0----------------------------
Try this example
```dart
@freezed
@Collection(ignore: {'copyWith'})
class Expense with _$Expense {
  const Expense._();

  @JsonSerializable()
  const factory Expense({
    required int id,
    required DateTime creationDate,
    required int amount,
    @JsonKey(name: 'name') required String note,
  }) = _Expense;

  @override
  // ignore: recursive_getters
  Id get id => id;

  factory Expense.fromJson(Map<String, dynamic> json) =>
      _$ExpenseFromJson(json);
}
```

**Reply by stevehayles** [reply]
Thanks to @masreplay for the example,  it was useful to know that was a working example.  In my case the issue was fixed by ensuring that that freezed build_runner ran before the isar_generator.

As soon as I added the following to my build.yaml file

```
global_options:
  freezed:freezed:
    runs_before:
      - isar_generator:isar_generator
```

My existing code  started working

**Reply by adnanjpg** [reply]
@stevehayles Thanks for the solution. Anyone using this solution with `isar` version 4 or higher, now the  `isar_generator` is included inside the `isar` package, so the builder is now called `isar:isar_generator`. so the addition to `build.yaml` should be as following:

```
global_options:
  freezed:freezed:
    runs_before:
      - isar:isar_generator
```

**Reply by arnab** [reply]
And for anyone who have started using the new https://github.com/isar-community/isar-community (via pub/dev), you need to change it to:

```yaml
global_options:
  freezed:freezed:
    runs_before:
      - isar_community_generator:isar_community_generator
```

_@vicenterusso for confirmation (and maybe we want to do add this to the README in the community fork)?_

**Comment by naumanmir** [comment]
Why isar doesn't support copyWith? It makes it incompatible with Freezed.
